lib/src/compile.ts

```diff
@@ -132,25 +132,24 @@ async function compileRequest(
     }
   );
 
-  dispatcher.error$.subscribe(error => {
-    throw error;
-  });
-  const response = await dispatcher.sendCompileRequest(request);
-  embeddedCompiler.close();
-
-  if (response.getSuccess()) {
-    const success = response.getSuccess()!;
-    const sourceMap = success.getSourceMap();
-    if (request.getSourceMap() && sourceMap === undefined) {
-      throw Error('Compiler did not provide sourceMap.');
+  try {
+    const response = await dispatcher.sendCompileRequest(request);
```

Comment:
@nex3 There is one more edge case here. If the dispatcher's underlying steam closes due to an error *before* sending this request, this `await` will hang. In this case, would it be better for the dispatcher to just resolve the `sendCompileRequest` promise, or reject it with an error?

Reply:
It should reject it with an error—a promise should only resolve successfully if its designated task has completed and it has a useful value to return.

Reply:
Right, done.